pkg/settings/setting.go

```diff
@@ -448,3 +449,19 @@ func IterateWhitelistedEnvVars(handler func(name, value string)) {
 		}
 	}
 }
+
+// GetMachineProvisionImagePullPolicy will return the pull policy to be used on MachineProvisioning job.
+// If an invalid value is set it will return the default value: v1.PullAlways
+func GetMachineProvisionImagePullPolicy() v1.PullPolicy {
+	switch v1.PullPolicy(MachineProvisionImagePullPolicy.Get()) {
```

Comment:
If this is the cleanest way to check if the passed value is one of supported values then I'm just going to rant that Go needs better support for enumerations. :(